core/src/test/java/io/lionweb/lioncore/java/model/impl/DynamicNodeTest.java

```diff
@@ -173,4 +170,64 @@ public void getRootCircularHierarchy() {
     assertThrows(IllegalStateException.class, () -> n3.getRoot());
     assertThrows(IllegalStateException.class, () -> n4.getRoot());
   }
+
+  @Test
+  public void settingFalseNonNullableBooleanProperty() {
+    Language l = new Language("MyLanguage", "l-id", "l-key", "123");
+    Concept a = new Concept(l, "A", "a-id", "a-key");
+    a.addFeature(Property.createRequired("foo", LionCoreBuiltins.getBoolean()));
+    DynamicNode n1 = new DynamicNode("n1", a);
+
+    assertEquals(false, n1.getPropertyValueByName("foo"));
+    n1.setPropertyValueByName("foo", false);
+    assertEquals(false, n1.getPropertyValueByName("foo"));
+  }
+
+  @Test
+  public void settingTrueNonNullableBooleanProperty() {
+    Language l = new Language("MyLanguage", "l-id", "l-key", "123");
+    Concept a = new Concept(l, "A", "a-id", "a-key");
+    a.addFeature(Property.createRequired("foo", LionCoreBuiltins.getBoolean()));
+    DynamicNode n1 = new DynamicNode("n1", a);
+
+    assertEquals(false, n1.getPropertyValueByName("foo"));
+    n1.setPropertyValueByName("foo", true);
+    assertEquals(true, n1.getPropertyValueByName("foo"));
+  }
+
+  @Test
+  public void settingFalseNullableBooleanProperty() {
+    Language l = new Language("MyLanguage", "l-id", "l-key", "123");
+    Concept a = new Concept(l, "A", "a-id", "a-key");
+    a.addFeature(Property.createOptional("foo", LionCoreBuiltins.getBoolean()));
+    DynamicNode n1 = new DynamicNode("n1", a);
+
+    assertEquals(null, n1.getPropertyValueByName("foo"));
+    n1.setPropertyValueByName("foo", false);
+    assertEquals(false, n1.getPropertyValueByName("foo"));
+  }
+
+  @Test
+  public void settingNullNullableBooleanProperty() {
+    Language l = new Language("MyLanguage", "l-id", "l-key", "123");
+    Concept a = new Concept(l, "A", "a-id", "a-key");
+    a.addFeature(Property.createOptional("foo", LionCoreBuiltins.getBoolean()));
+    DynamicNode n1 = new DynamicNode("n1", a);
+
+    assertEquals(null, n1.getPropertyValueByName("foo"));
+    n1.setPropertyValueByName("foo", null);
+    assertEquals(null, n1.getPropertyValueByName("foo"));
+  }
+
+  @Test
```

Comment:
I think we're missing two tests:

a) setting `null` to a non-nullable property
b) setting `null` to a nullable property that currently has a value

Reply:
I will add these tests, thank you

Reply:
Tests added directly in master, as I had already merged this